### DISCUSSION TITLE
Body dependency being serialized twice

### DISCUSSION THREAD

**Discussion by hootie-hoo** [fastapi/fastapi#11420] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post
async def create_blog(customer_id: Annotated[int, Depends(BlogCreatePermission())],
                       blog_request: Annotated[schemas.BlogRequest, Depends(validate_blog_post)])


class BlogCreatePermission:
    async def __call__(self,
                       auth: Annotated[Auth, Depends(Authorize())],
                       blog_request: schemas.BlogRequest):
        return await self.verify_permissions(auth, blog_request)

def validate_blog_post(blog_request: schemas.BlogRequest) -> schemas.BlogRequest
    ...
```

### Description

I am using the blog request body in two dependencies. The first checks for valid permission and then the second does some additional data validation. However, FastAPI is not reusing the blog_request class and in both the BlogCreatePermission call and validate_blog_post call I get a new instance of blog_request. Any ideas? From the docs it sounds like FastAPI should reuse the blog_post dependency.

### Operating System

macOS

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

3.10

**Comment by YuriiMotov** [comment]
Yes, this problem really exists.
This is a PR, that fixes this problem (but it hasn't merged yet): https://github.com/tiangolo/fastapi/pull/11323
